File: app/src/main/java/com/rendersoncs/report/view/login/RecoveryLoginActivity.java
package com.rendersoncs.report.view.login;

import android.content.Intent;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.rendersoncs.report.R;
import com.rendersoncs.report.infrastructure.util.CloseVirtualKeyBoardKt;

public class RecoveryLoginActivity extends AppCompatActivity implements TextView.OnEditorActionListener {

    private EditText email;
    private FirebaseAuth firebaseAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recovery_login);

        TextView returnLoginActivity = findViewById(R.id.sendHome);
        returnLoginActivity.setOnClickListener(view -> callLoginActivity());

        firebaseAuth = FirebaseAuth.getInstance();
        init();
    }

    private void callLoginActivity() {
        Intent i = (new Intent(this, LoginActivity.class));
        startActivity(i);
        finish();
    }

    private void init(){
        email = findViewById(R.id.emailRecover);
        email.setOnEditorActionListener(this);
    }

    public void reset( View view ){
        if (email.getText().toString().isEmpty()){
            Toast.makeText(this, getResources().getString(R.string.label_insert_email), Toast.LENGTH_SHORT).show();
        }else {
            firebaseAuth
                    .sendPasswordResetEmail(email.getText().toString())
                    .addOnCompleteListener(task -> {
                        if (task.isSuccessful()) {
                            email.setText("");
                            Toast.makeText(
                                    RecoveryLoginActivity.this,
                                    getResources().getString(R.string.label_recover_access_send),
                                    Toast.LENGTH_SHORT
                            ).show();
                        } else {
                            Toast.makeText(
                                    RecoveryLoginActivity.this,
                                    getResources().getString(R.string.label_failed),
                                    Toast.LENGTH_SHORT
                            ).show();
                        }
                    })
                    .addOnFailureListener(Throwable::printStackTrace);
        }
    }

    public boolean onEditorAction(TextView view, int actionId, KeyEvent event) {
        if (actionId == EditorInfo.IME_ACTION_DONE) {
            CloseVirtualKeyBoardKt.closeVirtualKeyBoard(this, view);
            this.reset(view);
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/rendersoncs/report/view/login/RecoveryLoginActivity.kt
package com.rendersoncs.report.view.login

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.FirebaseAuth
import com.rendersoncs.report.R
import com.rendersoncs.report.databinding.FragmentRecoveryLoginBinding
import com.rendersoncs.report.common.util.closeVirtualKeyBoard

class RecoveryLoginActivity : AppCompatActivity(), TextView.OnEditorActionListener {
    private var firebaseAuth: FirebaseAuth? = null
    private lateinit var binding: FragmentRecoveryLoginBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = FragmentRecoveryLoginBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        firebaseAuth = FirebaseAuth.getInstance()

        binding.emailRecover.setOnEditorActionListener(this)
        binding.emailRecover.addTextChangedListener(emailTextWatcher)
        binding.sendHome.setOnClickListener { callLoginActivity() }
    }

    private fun callLoginActivity() {
        finish()
    }

    private val emailTextWatcher = object : TextWatcher {
        override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}

        override fun onTextChanged(s: CharSequence?, p1: Int, p2: Int, p3: Int) {}

        override fun afterTextChanged(s: Editable?) {
            validatePassword(s)
        }
    }

    private fun validatePassword(s: Editable?) {
        if (s != null && s.isNotEmpty()) {
            binding.textInputEmailRecover.error = null
        } else {
            binding.textInputEmailRecover.error = getString(R.string.label_insert_email)
        }
    }

    fun btnRecoverAccount(view: View?) {
        recoveryAccount()
    }

    private fun recoveryAccount() {
        if (binding.emailRecover.text.toString().isEmpty()) {
            binding.textInputEmailRecover.error = getString(R.string.label_insert_email)
        } else {
            firebaseAuth
                    ?.sendPasswordResetEmail(binding.emailRecover.text.toString())
                    ?.addOnCompleteListener { task: Task<Void?> ->
                        if (task.isSuccessful) {
                            binding.emailRecover.setText("")
                            Toast.makeText(
                                    this@RecoveryLoginActivity,
                                    resources.getString(R.string.label_recover_access_send),
                                    Toast.LENGTH_SHORT
                            ).show()
                        } else {
                            Toast.makeText(
                                    this@RecoveryLoginActivity,
                                    resources.getString(R.string.label_failed),
                                    Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                    ?.addOnFailureListener { obj: Exception -> obj.printStackTrace() }
        }
    }

    override fun onEditorAction(view: TextView?, actionId: Int, keyEvent: KeyEvent?): Boolean {
        if (actionId == EditorInfo.IME_ACTION_DONE) {
            closeVirtualKeyBoard(
                    this,
                    view!!
            )
            this.recoveryAccount()
            return true
        }
        return false
    }
}